---
title: "Basal-like/Classical transcriptomic heatmap"
author: Astrid Deschênes
output:
    rmarkdown::html_document:
        highlight: pygments
        toc: true
        fig_width: 5
        number_sections: true
pkgdown:
  asis: true
bibliography: SignatureHeatmap.bibtex
vignette: >
    %\VignetteIndexEntry{TODO}
    %\VignettePackage{SignatureHeatmap}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'asis', warning=FALSE, message=FALSE}
BiocStyle::markdown()

suppressPackageStartupMessages({
  library(knitr)
  library(ComplexHeatmap)
  library(SignatureHeatmap)
})

set.seed(121)
```


<br />

**Package**: `r Rpackage("SignatureHeatmap")`<br />
**Authors**: `r packageDescription("SignatureHeatmap")[["Author"]]`<br />
**Version**: `r packageDescription("SignatureHeatmap")$Version`<br />
**Compiled date**: `r Sys.Date()`<br />
**License**: `r packageDescription("SignatureHeatmap")[["License"]]`<br />


# Licensing 


The `r Githubpkg("adeschen/SignatureHeatmap")` package and the underlying 
`r Githubpkg("adeschen/SignatureHeatmap")` code are distributed under the 
Artistic license 2.0. You are free to use and redistribute this software. 

<br>
<br>


# Citing


If you use the `r Githubpkg("adeschen/SignatureHeatmap")` package for 
a publication, we would ask you to cite the following:

> Deschênes A, King D and Tuveson DA. SignatureHeatmap: Heatmap using basal-like and classical gene list and RNA-seq expression data. R package version 0.0.2, https://adeschen.github.io/SignatureHeatmap/ 
  
You can also replace it by the Research Resource Identifier __(RRID)__, as 
example:

>   SignatureHeatmap version 0.0.3, RRID: SCR_022949

If you use one of the available Basal-like/Classical gene list, the appropriate
paper should be referenced.

For Moffitt et al 2015

> Moffitt RA, Marayati R, Flate EL et al. Virtual microdissection identifies distinct tumor- and stroma-specific subtypes of pancreatic ductal adenocarcinoma. Nat Genet. 2015 Oct;47(10):1168-78. doi: 10.1038/ng.3398. Epub 2015 Sep 7. PMID: 26343385; PMCID: PMC4912058.

For Tiriac et al. 2018

> Tiriac H, Belleau P, Engle DD et al. Organoid Profiling Identifies Common Responders to Chemotherapy in Pancreatic Cancer. Cancer Discov. 2018 Sep;8(9):1112-1129. doi: 10.1158/2159-8290.CD-18-0349. Epub 2018 May 31. PMID: 29853643; PMCID: PMC6125219.

For Miyabayashi et al. 2020

> Miyabayashi K, Baker LA, Deschênes A et al. Intraductal Transplantation Models of Human Pancreatic Ductal Adenocarcinoma Reveal Progressive Transition of Molecular Subtypes. Cancer Discov. 2020 Oct;10(10):1566-1589. doi: 10.1158/2159-8290.CD-20-0133. Epub 2020 Jul 23. PMID: 32703770; PMCID: PMC7664990.

<br>
<br>

# Introduction

Pancreatic ductal adenocarcinoma (PDAC) is an aggressive form of cancer. 
Bulk transcriptomic analyses distinguished between two main classes 
of tumors: _classical_ and _basal-like_ . The 
_basal-like_ tumors show a more aggressive clinical behavior [@Moffitt2015].

This package enables Basal-like/Classical classification of 
pancreatic tumor samples by generating a heatmap based on transcriptomic data. 
There is three different lists of genes available in the package: the original 
list from Moffitt et al. 2015 [@Moffitt2015]; the updated list 
from Miyabayashi et al. 2020 [@Miyabayashi2020] and the list from 
Tiriac et al. 2018 [@Tiriac2018] that has been developed for pancreatic ductal 
adenocarcinoma (PDAC) organoids.

<br>
<br>

# Installation

The [devtools](https://cran.r-project.org/web/packages/devtools/index.html) 
package is required to install this package.

You need to start R (version "4.2" or higher) and enter: 

```{r installDemo01, eval=FALSE}
if (!requireNamespace("devtools", quietly = TRUE))
  install.packages("devtools")

## Install the latest version of SignatureHeatmap
 devtools::install_github('adeschen/SignatureHeatmap')
```


<br>
<br>

# Input data

The function _createHeatmap()_ is needed to generate the classification 
heatmap. This function required:

1. A _data.frame_ containing transcriptomic expression
2. A _data.frame_ containing the Basal-like/Classical genes


## Transcriptomic expression input

The transcriptomic expression must be in a _data.frame_ format and 
have one column per sample and one column 
with the name of the genes represented in the rows. 

A demo dataset called _demo_PDAC_TCGA_ is available in the 
`r Githubpkg("adeschen/SignatureHeatmap")` package. This dataset is 
composed of few pancreatic ductal adenocarcinoma samples from TCGA. The 
column for the gene names is called '_GENE_'.

```{r demoRNA, eval=TRUE}

data("demo_PDAC_TCGA")

head(demo_PDAC_TCGA[, (ncol(demo_PDAC_TCGA)-4):ncol(demo_PDAC_TCGA)], nrow=3)

```

<br>
<br>

## Basal-like/Classical classification input


There are 3 Basal-like/Classical gene lists available in the 
`r Githubpkg("adeschen/SignatureHeatmap")` package:

- Moffitt et al 2015
- Tiriac et al 2018
- Miyabayashi et al 2020


The Moffitt et al 2015 list corresponds to the original Basal-like/Classical 
classification gene list [@Moffitt2015]. The list contains 50 genes: 
25 Basal-like and 25 Classical genes.

The Tiriac et al 2018 list has been developed to classify PDAC 
organoids [@Tiriac2018]. This list contains 88 genes: 22 Basal-like and 
62 Classical genes.

The Miyabayashi et al 2020 is an update of the original Moffitt 
Basal-like/Classical classification gene list [@Miyabayashi2020]. Gene symbols 
that were not found anymore in most recent versions of Ensembl have been
replaced with their newest symbols. As the Moffitt gene list, this list 
contains 25 Basal-like and 25 Classical genes.

You can use you own gene list as long as the genes are in a _data.frame_ 
containing those 2 columns: '_SYMBOL_' and '_Class_'. The '_Class_' column 
contains the classification formatted as 'Basal-like' and 'Classical' (with 
respect to upper letters).

```{r classificationDataset, eval=TRUE}
## Moffitt et al 2015
data("Moffitt_2015")

## Tiriac et al 2018
data("Tiriac_2018")

## Miyabayashi et al 2020
data("Miyabayashi_2020")

## Each dataset contains 2 columns
## The first column has the gene symbols
## The second column the class assignation (Basal-like or Classical)
head(Moffitt_2015)

```

<br>
<br>

# Create heatmap

The  _createHeatmap()_ function is needed to generate the classification 
heatmap. 

This first example uses a dataset composed of few pancreatic ductal 
adenocarcinoma samples from TCGA. It shows the heatmap generated with 
the default options.

```{r demo01, eval=TRUE, fig.height=8, fig.align="center"}
## First, fix the seed value
set.seed(142)

## Load Moffitt et al 2015 gene list
data("Moffitt_2015")

## Load demo PDAC TCGA expression data
data("demo_PDAC_TCGA")

## Create heatmap with default parameters
## The column for the gene names in the rna_data data.frame has 
## the default name "GENE"
createHeatmap(gene_list=Moffitt_2015, rna_data=demo_PDAC_TCGA)
```

<br>
<br>

The _createHeatmap()_ function encapsulates the _ComplexHeatmap::Heatmap_ 
function. All parameters, except __right_annotation__, from 
_ComplexHeatmap::Heatmap_ function can be passed to the _createHeatmap()_ 
function.  

```{r demo02, eval=TRUE, fig.height=6, fig.align="center"}
## Library needed to use gpar() function
library(grid)

## First, fix the seed value
set.seed(122)

## Load Moffitt et al 2015 gene list
data("Moffitt_2015")

## Load demo PDAC TCGA expression data
data("demo_PDAC_TCGA")

## Create heatmap with personalized parameters
## The column for the gene names in the rna_data data.frame 
##     has the default name "GENE"
## The clustering uses Spearman method
## The rows and columns are split in 2 groups
## The row names are not shown
createHeatmap(gene_list=Moffitt_2015, rna_data=demo_PDAC_TCGA,  
    clustering_distance_rows = "spearman", 
    clustering_distance_columns="spearman", 
    row_split=2, column_split=2, column_names_gp = gpar(fontsize = 8),
    show_row_names=FALSE)
```

<br>
<br>

The output of the _createHeatmap()_ function is a 
__ComplexHeatmap::Heatmap-class__ object that can be used as a parameter in 
the __draw()__ function to enable further personalization of the graph.

```{r demo03, eval=TRUE, fig.height=6, fig.align="center"}
## Required library
library(ComplexHeatmap)

## First, fix the seed value
set.seed(188)

## Load Moffitt et al 2015 gene list
data("Moffitt_2015")

## Load demo PDAC TCGA expression data
data("demo_PDAC_TCGA")

## Create heatmap 
## The column for the gene names in the rna_data data.frame 
##     has the default name "GENE"
gg <- createHeatmap(gene_list=Moffitt_2015, rna_data=demo_PDAC_TCGA,  
        clustering_distance_rows = "spearman", 
        clustering_distance_columns="spearman", 
        row_split=2, column_split=2, show_row_names=FALSE, 
        show_column_names=FALSE)

## Assign row and column titles
uu <- draw(gg, show_heatmap_legend=FALSE, 
        column_title = "Samples", row_title="Basal-like/Classical Genes", 
        column_title_gp=gpar(fontsize="12", fontface="bold"), 
        row_title_gp=gpar(fontsize="12", fontface="bold"))
```

<br>
<br>


# Reproducible research

To ensure reproducible results, the *set.seed()* function should be call 
before calling the *createHeatmap()* function. 

As example, a first graph can be used to visually identify the 
basal-like and classical samples.

```{r demoSeed, eval=TRUE, fig.height=6, fig.align="center"}
## Load Miyabayashi et al 2020 gene list
data("Miyabayashi_2020")

## Load demo PDAC TCGA expression data
data("demo_PDAC_TCGA")

## Fix the seed value to ensure reproducible results
set.seed(144)

## Create heatmap
newGraph <- createHeatmap(gene_list=Miyabayashi_2020, rna_data=demo_PDAC_TCGA, 
    row_split=2, column_split=2, show_row_names=FALSE, show_column_names=FALSE)

newGraph
```

<br>

A second identical graph can be generated by fixing the `seed`. That second 
graph can be updated with the Moffitt assignment for the sample.

```{r demoSeed2, warning=FALSE, eval=TRUE, fig.height=6, fig.align="center"}
library(ComplexHeatmap)

## Identify the group associated to a specific subtype
## According to graph, the second group contains the classical samples
infoColumn <- column_order(newGraph)
classical <- colnames(newGraph@matrix)[infoColumn[[2]]]

## Generate data.frame containing the subtype for each sample
subtype <- data.frame(Sample=c(colnames(newGraph@matrix)),
              Subtype=c(rep("Basal-like", length(colnames(newGraph@matrix)))))
subtype$Subtype[subtype$Sample %in% classical] <- "Classical"
rownames(subtype) <- subtype$Sample

## Create a column annotation
colAnnotation <- HeatmapAnnotation(Subtype=subtype[, "Subtype"], 
      col=list(Subtype=c("Classical"="darkviolet", "Basal-like"="darkorange")),
      show_legend = FALSE)

## First, fix the seed value to the same value ensure reproducible results
set.seed(144)

createHeatmap(gene_list=Miyabayashi_2020, rna_data=demo_PDAC_TCGA, 
  row_split=2, column_split=2, show_row_names=FALSE, show_column_names=FALSE,
  top_annotation=colAnnotation)

```


# Acknowledgments

This work was supported by the Lustgarten Foundation, where David A. Tuveson 
is a distinguished scholar and Director of the Lustgarten Foundation–designated 
Laboratory of Pancreatic Cancer Research. 

The results shown here are in part based upon data generated by the 
TCGA Research Network: https://www.cancer.gov/tcga.

<br>
<br>


# Session info

Here is the output of `sessionInfo()` on the system on which this document was 
compiled:

```{r sessionInfo, echo=FALSE}
sessionInfo()
```

<br>
<br>

# References

