<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Basal-like/Classical transcriptomic heatmap • BasalClassicalHeatmap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Basal-like/Classical transcriptomic heatmap">
<meta property="og:description" content="BasalClassicalHeatmap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BasalClassicalHeatmap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/BasalClassicalHeatmap.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/adeschen/BasalClassicalHeatmap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Basal-like/Classical transcriptomic heatmap</h1>
                        <h4 data-toc-skip class="author">Astrid
Deschênes</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/adeschen/BasalClassicalHeatmap/blob/HEAD/vignettes/BasalClassicalHeatmap.Rmd" class="external-link"><code>vignettes/BasalClassicalHeatmap.Rmd</code></a></small>
      <div class="hidden name"><code>BasalClassicalHeatmap.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><p><br></p>
<p><strong>Package</strong>: <em>BasalClassicalHeatmap</em><br><strong>Authors</strong>: Astrid Deschênes [aut, cre] (<a href="https://orcid.org/0000-0001-7846-6749" class="external-link uri">https://orcid.org/0000-0001-7846-6749</a>), Daniel King
[dtc], David A. Tuveson [fnd] (<a href="https://orcid.org/0000-0002-8017-2712" class="external-link uri">https://orcid.org/0000-0002-8017-2712</a>)<br><strong>Version</strong>: 0.0.4<br><strong>Compiled date</strong>:
2022-12-15<br><strong>License</strong>: Artistic-2.0<br></p>
<div class="section level2">
<h2 id="licensing">Licensing<a class="anchor" aria-label="anchor" href="#licensing"></a>
</h2>
<p>The <em><a href="https://github.com/adeschen/BasalClassicalHeatmap" class="external-link">BasalClassicalHeatmap</a></em>
package and the underlying <em><a href="https://github.com/adeschen/BasalClassicalHeatmap" class="external-link">BasalClassicalHeatmap</a></em>
code are distributed under the Artistic license 2.0. You are free to use
and redistribute this software.</p>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="citing">Citing<a class="anchor" aria-label="anchor" href="#citing"></a>
</h2>
<p>If you use the <em><a href="https://github.com/adeschen/BasalClassicalHeatmap" class="external-link">BasalClassicalHeatmap</a></em>
package for a publication, we would ask you to cite the following:</p>
<blockquote>
<p>Deschênes A, King D and Tuveson DA. BasalClassicalHeatmap: Heatmap
using basal-like and classical gene list and RNA-seq expression data. R
package version 0.0.2, <a href="https://adeschen.github.io/BasalClassicalHeatmap/" class="external-link uri">https://adeschen.github.io/BasalClassicalHeatmap/</a></p>
</blockquote>
<p>You can also replace it by the Research Resource Identifier
<strong>(RRID)</strong>, as example:</p>
<blockquote>
<p>BasalClassicalHeatmap version 0.0.2, RRID: SCR_022949</p>
</blockquote>
<p>If you use one of the available Basal-like/Classical gene list, the
appropriate paper should be referenced.</p>
<p>For Moffitt et al 2015</p>
<blockquote>
<p>Moffitt RA, Marayati R, Flate EL et al. Virtual microdissection
identifies distinct tumor- and stroma-specific subtypes of pancreatic
ductal adenocarcinoma. Nat Genet. 2015 Oct;47(10):1168-78. doi:
10.1038/ng.3398. Epub 2015 Sep 7. PMID: 26343385; PMCID: PMC4912058.</p>
</blockquote>
<p>For Tiriac et al. 2018</p>
<blockquote>
<p>Tiriac H, Belleau P, Engle DD et al. Organoid Profiling Identifies
Common Responders to Chemotherapy in Pancreatic Cancer. Cancer Discov.
2018 Sep;8(9):1112-1129. doi: 10.1158/2159-8290.CD-18-0349. Epub 2018
May 31. PMID: 29853643; PMCID: PMC6125219.</p>
</blockquote>
<p>For Miyabayashi et al. 2020</p>
<blockquote>
<p>Miyabayashi K, Baker LA, Deschênes A et al. Intraductal
Transplantation Models of Human Pancreatic Ductal Adenocarcinoma Reveal
Progressive Transition of Molecular Subtypes. Cancer Discov. 2020
Oct;10(10):1566-1589. doi: 10.1158/2159-8290.CD-20-0133. Epub 2020 Jul
23. PMID: 32703770; PMCID: PMC7664990.</p>
</blockquote>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Pancreatic ductal adenocarcinoma (PDAC) is an aggressive form of
cancer. Bulk transcriptomic analyses distinguished between two main
classes of tumors: <em>classical</em> and <em>basal-like</em> . The
<em>basal-like</em> tumors show a more aggressive clinical behavior
<span class="citation">(Moffitt et al. 2015)</span>.</p>
<p>This package enables Basal-like/Classical classification of
pancreatic tumor samples by generating a heatmap based on transcriptomic
data. There is three different lists of genes available in the package:
the original list from Moffitt et al. 2015 <span class="citation">(Moffitt et al. 2015)</span>; the updated list from
Miyabayashi et al. 2020 <span class="citation">(Miyabayashi et al.
2020)</span> and the list from Tiriac et al. 2018 <span class="citation">(Tiriac et al. 2018)</span> that has been developed for
pancreatic ductal adenocarcinoma (PDAC) organoids.</p>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The <a href="https://cran.r-project.org/web/packages/devtools/index.html" class="external-link">devtools</a>
package is required to install this package.</p>
<p>You need to start R (version “4.2” or higher) and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Install the latest version of BasalClassicalHeatmap</span></span>
<span> <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'adeschen/BasalClassicalHeatmap'</span><span class="op">)</span></span></code></pre></div>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>The function <em>createHeatmap()</em> is needed to generate the
classification heatmap. This function required:</p>
<ol style="list-style-type: decimal">
<li>A <em>data.frame</em> containing transcriptomic expression</li>
<li>A <em>data.frame</em> containing the Basal-like/Classical genes</li>
</ol>
<div class="section level3">
<h3 id="transcriptomic-expression-input">Transcriptomic expression input<a class="anchor" aria-label="anchor" href="#transcriptomic-expression-input"></a>
</h3>
<p>The transcriptomic expression must be in a <em>data.frame</em> format
and have one column per sample and one column with the name of the genes
represented in the rows.</p>
<p>A demo dataset called <em>demo_PDAC_TCGA</em> is available in the
<em><a href="https://github.com/adeschen/BasalClassicalHeatmap" class="external-link">BasalClassicalHeatmap</a></em>
package. This dataset is composed of few pancreatic ductal
adenocarcinoma samples from TCGA. The column for the gene names is
called ‘<em>GENE</em>’.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"demo_PDAC_TCGA"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">demo_PDAC_TCGA</span><span class="op">[</span>, <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">demo_PDAC_TCGA</span><span class="op">)</span><span class="op">-</span><span class="fl">4</span><span class="op">)</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">demo_PDAC_TCGA</span><span class="op">)</span><span class="op">]</span>, nrow<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         TCGA-HZ-8636-01A-21R-2404-07 TCGA-IB-A6UG-01A-32R-A33R-07</span></span>
<span><span class="co">## ADORA2B                 4.668218e-06                 3.443706e-06</span></span>
<span><span class="co">## AGR2                    1.351521e-03                 1.840877e-04</span></span>
<span><span class="co">## AGR3                    6.481299e-05                 1.278232e-05</span></span>
<span><span class="co">## AKR1C3                  1.040258e-04                 3.881818e-05</span></span>
<span><span class="co">## AMIGO2                  3.257307e-05                 1.032804e-04</span></span>
<span><span class="co">## ANO1                    1.190121e-04                 2.996840e-05</span></span>
<span><span class="co">##         TCGA-2J-AABO-01A-21R-A41B-07 TCGA-IB-A5SS-01A-11R-A32O-07    GENE</span></span>
<span><span class="co">## ADORA2B                 2.258423e-06                 1.399798e-05 ADORA2B</span></span>
<span><span class="co">## AGR2                    1.166542e-03                 4.845146e-04    AGR2</span></span>
<span><span class="co">## AGR3                    1.142631e-04                 7.797597e-05    AGR3</span></span>
<span><span class="co">## AKR1C3                  1.340612e-04                 5.114997e-05  AKR1C3</span></span>
<span><span class="co">## AMIGO2                  6.070233e-05                 6.050099e-05  AMIGO2</span></span>
<span><span class="co">## ANO1                    1.047437e-04                 1.183015e-04    ANO1</span></span></code></pre>
<p><br><br></p>
</div>
<div class="section level3">
<h3 id="basal-likeclassical-classification-input">Basal-like/Classical classification input<a class="anchor" aria-label="anchor" href="#basal-likeclassical-classification-input"></a>
</h3>
<p>There are 3 Basal-like/Classical gene lists available in the <em><a href="https://github.com/adeschen/BasalClassicalHeatmap" class="external-link">BasalClassicalHeatmap</a></em>
package:</p>
<ul>
<li>Moffitt et al 2015</li>
<li>Tiriac et al 2018</li>
<li>Miyabayashi et al 2020</li>
</ul>
<p>The Moffitt et al 2015 list corresponds to the original
Basal-like/Classical classification gene list <span class="citation">(Moffitt et al. 2015)</span>. The list contains 50
genes: 25 Basal-like and 25 Classical genes.</p>
<p>The Tiriac et al 2018 list has been developed to classify PDAC
organoids <span class="citation">(Tiriac et al. 2018)</span>. This list
contains 88 genes: 22 Basal-like and 62 Classical genes.</p>
<p>The Miyabayashi et al 2020 is an update of the original Moffitt
Basal-like/Classical classification gene list <span class="citation">(Miyabayashi et al. 2020)</span>. Gene symbols that
were not found anymore in most recent versions of Ensembl have been
replaced with their newest symbols. As the Moffitt gene list, this list
contains 25 Basal-like and 25 Classical genes.</p>
<p>You can use you own gene list as long as the genes are in a
<em>data.frame</em> containing those 2 columns: ‘<em>SYMBOL</em>’ and
‘<em>Class</em>’. The ‘<em>Class</em>’ column contains the
classification formatted as ‘Basal-like’ and ‘Classical’ (with respect
to upper letters).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Moffitt et al 2015</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Moffitt_2015"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Tiriac et al 2018</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Tiriac_2018"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Miyabayashi et al 2020</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Miyabayashi_2020"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Each dataset contains 2 columns</span></span>
<span><span class="co">## The first column has the gene symbols</span></span>
<span><span class="co">## The second column the class assignation (Basal-like or Classical)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Moffitt_2015</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   SYMBOL      Class</span></span>
<span><span class="co">## 1  VGLL1 Basal-like</span></span>
<span><span class="co">## 2   UCA1 Basal-like</span></span>
<span><span class="co">## 3 S100A2 Basal-like</span></span>
<span><span class="co">## 4   LY6D Basal-like</span></span>
<span><span class="co">## 5  SPRR3 Basal-like</span></span>
<span><span class="co">## 6 SPRR1B Basal-like</span></span></code></pre>
<p><br><br></p>
</div>
</div>
<div class="section level2">
<h2 id="create-heatmap">Create heatmap<a class="anchor" aria-label="anchor" href="#create-heatmap"></a>
</h2>
<p>The <em>createHeatmap()</em> function is needed to generate the
classification heatmap.</p>
<p>This first example uses a dataset composed of few pancreatic ductal
adenocarcinoma samples from TCGA. It shows the heatmap generated with
the default options.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## First, fix the seed value</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">142</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load Moffitt et al 2015 gene list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Moffitt_2015"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load demo PDAC TCGA expression data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"demo_PDAC_TCGA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create heatmap with default parameters</span></span>
<span><span class="co">## The column for the gene names in the rna_data data.frame has </span></span>
<span><span class="co">## the default name "GENE"</span></span>
<span><span class="fu"><a href="../reference/createHeatmap.html">createHeatmap</a></span><span class="op">(</span>gene_list<span class="op">=</span><span class="va">Moffitt_2015</span>, rna_data<span class="op">=</span><span class="va">demo_PDAC_TCGA</span><span class="op">)</span></span></code></pre></div>
<p><img src="BasalClassicalHeatmap_files/figure-html/demo01-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br><br></p>
<p>The <em>createHeatmap()</em> function encapsulates the
<em>ComplexHeatmap::Heatmap</em> function. All parameters, except
<strong>right_annotation</strong>, from <em>ComplexHeatmap::Heatmap</em>
function can be passed to the <em>createHeatmap()</em> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Library needed to use gpar() function</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## First, fix the seed value</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">122</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load Moffitt et al 2015 gene list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Moffitt_2015"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load demo PDAC TCGA expression data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"demo_PDAC_TCGA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create heatmap with personalized parameters</span></span>
<span><span class="co">## The column for the gene names in the rna_data data.frame </span></span>
<span><span class="co">##     has the default name "GENE"</span></span>
<span><span class="co">## The clustering uses Spearman method</span></span>
<span><span class="co">## The rows and columns are split in 2 groups</span></span>
<span><span class="co">## The row names are not shown</span></span>
<span><span class="fu"><a href="../reference/createHeatmap.html">createHeatmap</a></span><span class="op">(</span>gene_list<span class="op">=</span><span class="va">Moffitt_2015</span>, rna_data<span class="op">=</span><span class="va">demo_PDAC_TCGA</span>,  </span>
<span>    clustering_distance_rows <span class="op">=</span> <span class="st">"spearman"</span>, </span>
<span>    clustering_distance_columns<span class="op">=</span><span class="st">"spearman"</span>, </span>
<span>    row_split<span class="op">=</span><span class="fl">2</span>, column_split<span class="op">=</span><span class="fl">2</span>, column_names_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>    show_row_names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="BasalClassicalHeatmap_files/figure-html/demo02-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br><br></p>
<p>The output of the <em>createHeatmap()</em> function is a
<strong>ComplexHeatmap::Heatmap-class</strong> object that can be used
as a parameter in the <strong>draw()</strong> function to enable further
personalization of the graph.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Required library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## First, fix the seed value</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">188</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load Moffitt et al 2015 gene list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Moffitt_2015"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load demo PDAC TCGA expression data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"demo_PDAC_TCGA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create heatmap </span></span>
<span><span class="co">## The column for the gene names in the rna_data data.frame </span></span>
<span><span class="co">##     has the default name "GENE"</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createHeatmap.html">createHeatmap</a></span><span class="op">(</span>gene_list<span class="op">=</span><span class="va">Moffitt_2015</span>, rna_data<span class="op">=</span><span class="va">demo_PDAC_TCGA</span>,  </span>
<span>        clustering_distance_rows <span class="op">=</span> <span class="st">"spearman"</span>, </span>
<span>        clustering_distance_columns<span class="op">=</span><span class="st">"spearman"</span>, </span>
<span>        row_split<span class="op">=</span><span class="fl">2</span>, column_split<span class="op">=</span><span class="fl">2</span>, show_row_names<span class="op">=</span><span class="cn">FALSE</span>, </span>
<span>        show_column_names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Assign row and column titles</span></span>
<span><span class="va">uu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">gg</span>, show_heatmap_legend<span class="op">=</span><span class="cn">FALSE</span>, </span>
<span>        column_title <span class="op">=</span> <span class="st">"Samples"</span>, row_title<span class="op">=</span><span class="st">"Basal-like/Classical Genes"</span>, </span>
<span>        column_title_gp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="st">"12"</span>, fontface<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>, </span>
<span>        row_title_gp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="st">"12"</span>, fontface<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="BasalClassicalHeatmap_files/figure-html/demo03-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="reproducible-research">Reproducible research<a class="anchor" aria-label="anchor" href="#reproducible-research"></a>
</h2>
<p>To ensure reproducible results, the <em>set.seed()</em> function
should be call before calling the <em>createHeatmap()</em> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## First, fix the seed value to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">144</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load Miyabayashi et al 2020 gene list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Miyabayashi_2020"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load demo PDAC TCGA expression data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"demo_PDAC_TCGA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create heatmap</span></span>
<span><span class="fu"><a href="../reference/createHeatmap.html">createHeatmap</a></span><span class="op">(</span>gene_list<span class="op">=</span><span class="va">Miyabayashi_2020</span>, rna_data<span class="op">=</span><span class="va">demo_PDAC_TCGA</span>, </span>
<span>    row_split<span class="op">=</span><span class="fl">2</span>, column_split<span class="op">=</span><span class="fl">2</span>, show_row_names<span class="op">=</span><span class="cn">FALSE</span>, show_column_names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="BasalClassicalHeatmap_files/figure-html/demoSeed-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>This work was supported by the Lustgarten Foundation, where David A.
Tuveson is a distinguished scholar and Director of the Lustgarten
Foundation–designated Laboratory of Pancreatic Cancer Research.</p>
<p>The results shown here are in part based upon data generated by the
TCGA Research Network: <a href="https://www.cancer.gov/tcga" class="external-link uri">https://www.cancer.gov/tcga</a>.</p>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled:</p>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BasalClassicalHeatmap_0.0.4 ComplexHeatmap_2.12.1      </span></span>
<span><span class="co">## [3] knitr_1.41                  BiocStyle_2.24.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] circlize_0.4.15     shape_1.4.6         GetoptLong_1.0.5   </span></span>
<span><span class="co">##  [4] xfun_0.35           bslib_0.4.1         purrr_0.3.4        </span></span>
<span><span class="co">##  [7] colorspace_2.0-3    vctrs_0.5.1         htmltools_0.5.4    </span></span>
<span><span class="co">## [10] stats4_4.2.1        yaml_2.3.6          rlang_1.0.6        </span></span>
<span><span class="co">## [13] pkgdown_2.0.6       jquerylib_0.1.4     glue_1.6.2         </span></span>
<span><span class="co">## [16] BiocGenerics_0.42.0 RColorBrewer_1.1-3  matrixStats_0.63.0 </span></span>
<span><span class="co">## [19] foreach_1.5.2       lifecycle_1.0.3     stringr_1.5.0      </span></span>
<span><span class="co">## [22] ragg_1.2.2          GlobalOptions_0.1.2 codetools_0.2-18   </span></span>
<span><span class="co">## [25] memoise_2.0.1       evaluate_0.19       IRanges_2.30.1     </span></span>
<span><span class="co">## [28] fastmap_1.1.0       doParallel_1.0.17   parallel_4.2.1     </span></span>
<span><span class="co">## [31] highr_0.9           som_0.3-5.1         BiocManager_1.30.19</span></span>
<span><span class="co">## [34] cachem_1.0.6        desc_1.4.2          S4Vectors_0.34.0   </span></span>
<span><span class="co">## [37] jsonlite_1.8.4      systemfonts_1.0.4   fs_1.5.2           </span></span>
<span><span class="co">## [40] textshaping_0.3.6   rjson_0.2.21        png_0.1-8          </span></span>
<span><span class="co">## [43] digest_0.6.31       stringi_1.7.8       rprojroot_2.0.3    </span></span>
<span><span class="co">## [46] clue_0.3-63         cli_3.4.1           tools_4.2.1        </span></span>
<span><span class="co">## [49] magrittr_2.0.3      sass_0.4.4          cluster_2.1.3      </span></span>
<span><span class="co">## [52] crayon_1.5.2        rmarkdown_2.18      iterators_1.0.14   </span></span>
<span><span class="co">## [55] R6_2.5.1            compiler_4.2.1</span></span></code></pre>
<p><br><br></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Miyabayashi2020" class="csl-entry">
Miyabayashi, Koji, Lindsey A. Baker, Astrid Deschênes, Benno Traub,
Giuseppina Caligiuri, Dennis Plenker, Brinda Alagesan, et al. 2020.
<span>“<span class="nocase">Intraductal Transplantation Models of Human
Pancreatic Ductal Adenocarcinoma Reveal Progressive Transition of
Molecular Subtypes</span>.”</span> <em>Cancer Discovery</em>, no.
OctOber. <a href="https://doi.org/10.1158/2159-8290.cd-20-0133" class="external-link">https://doi.org/10.1158/2159-8290.cd-20-0133</a>.
</div>
<div id="ref-Moffitt2015" class="csl-entry">
Moffitt, R A, R Marayati, E L Flate, K E Volmar, S G Loeza, K A Hoadley,
N U Rashid, et al. 2015. <span>“<span class="nocase">Virtual
microdissection identifies distinct tumor- and stroma-specific subtypes
of pancreatic ductal adenocarcinoma</span>.”</span> <em>Nat Genet</em>
47 (10): 1168–78. <a href="https://doi.org/10.1038/ng.3398" class="external-link">https://doi.org/10.1038/ng.3398</a>.
</div>
<div id="ref-Tiriac2018" class="csl-entry">
Tiriac, Hervé, Pascal Belleau, Dannielle D Engle, Dennis Plenker, Astrid
Deschênes, Tim Somerville, Fieke E. M. Froeling, et al. 2018.
<span>“<span class="nocase">Organoid profiling identifies common
responders to chemotherapy in pancreatic cancer</span>.”</span>
<em>Cancer Discovery</em>, CD-18-0349. <a href="https://doi.org/10.1158/2159-8290.cd-18-0349" class="external-link">https://doi.org/10.1158/2159-8290.cd-18-0349</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Astrid Deschênes, David A. Tuveson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
